### PR TITLE
Implement sharing to a new note (#56)

### PR DESCRIPTION
**Pull request by trubitsyn**
Allows to create a new note with given text from other apps with title set using a dialog.

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt [hunk 1]
```diff
@@ -48,6 +48,17 @@ class MainActivity : SimpleActivity(), ViewPager.OnPageChangeListener {
...
+        with(intent) {
+            if (action == Intent.ACTION_SEND && type != null) {
+                if (type.equals("text/plain")) {
+                    val text = getStringExtra(Intent.EXTRA_TEXT)
+                    text?.let {
+                        displayNewNoteDialog(it)
+                    }
+                }
+            }
+        }
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt]: any particular difference between "with(intent)" and "intent.apply"? Im using apply everywhere. Also you can remove the local "text" variable and change the type check to 
` if (action == Intent.ACTION_SEND && type?.equals("text/plain") == true) {`

Comment (trubitsyn) [inline app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt]: Local "text" variable might be null, so it is needed. I'm unsure about `type?.equals("text/plain")`. Isn't `type == "text/plain"`enough?

Comment (trubitsyn) [inline app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt]: I personally don't like `getStringExtra(Intent.EXTRA_TEXT)?.let {
                    displayNewNoteDialog(it)
                }`. It feels cluttered.

Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/notes/activities/MainActivity.kt]: not sure about type, Id have to test it. You can also check the intenr.extra_text with `contains`, then the nullability isnt needed at all

**Comment by trubitsyn** [comment]
I've found a strange bug with this patch.
1. Receive any text
2. Cancel dialog
3. Change activity (switch to Settings)
4. Return to the main screen using back button
5. Change device orientation
6. Dialog reappears

Sorry for that, changing PR status to WIP.

**Comment by trubitsyn** [comment]
Fixed that. I've also moved the `"text/plain"` string to Constants class for a possible reuse.

**Comment by tibbi** [comment]
thanks, ive just made intent not nullable, it should never be null

**Pull request merged**